plugins/webp-uploads/tests/test-picture-element.php

```diff
@@ -125,4 +140,124 @@ public function data_provider_it_should_maybe_wrap_images_in_picture_element():
 			),
 		);
 	}
+
+	public function test_that_img_sizes_is_equal_to_source_sizes_for_picture_element(): void {
+		$mime_type = 'image/webp';
+		if ( ! wp_image_editor_supports( array( 'mime_type' => $mime_type ) ) ) {
+			$this->markTestSkipped( "Mime type $mime_type is not supported." );
+		}
+
+		$image = wp_get_attachment_image(
+			self::$image_id,
+			'large',
+			false,
+			array(
+				'class' => 'wp-image-' . self::$image_id,
+				'alt'   => 'Green Leaves',
+			)
+		);
+
+		$img_markup = apply_filters( 'the_content', $image );
+
+		// Apply picture element support.
+		$this->opt_in_to_picture_element();
+
+		$picture_markup = apply_filters( 'the_content', $image );
+
+		$img_processor = new WP_HTML_Tag_Processor( $img_markup );
+		$img_processor->next_tag( array( 'tag_name' => 'IMG' ) );
+		$img_sizes = $img_processor->get_attribute( 'sizes' );
+
+		$picture_processor = new WP_HTML_Tag_Processor( $picture_markup );
+		$picture_processor->next_tag( array( 'tag_name' => 'source' ) );
+		$picture_source_sizes = $picture_processor->get_attribute( 'sizes' );
+
+		$this->assertSame( $img_sizes, $picture_source_sizes );
+	}
+
+	public function test_that_img_sizes_is_equal_to_source_sizes_for_after_new_sizes_filter_picture_element(): void {
+		$mime_type = 'image/webp';
+		if ( ! wp_image_editor_supports( array( 'mime_type' => $mime_type ) ) ) {
+			$this->markTestSkipped( "Mime type $mime_type is not supported." );
+		}
+
+		$image = wp_get_attachment_image(
+			self::$image_id,
+			'large',
+			false,
+			array(
+				'class' => 'wp-image-' . self::$image_id,
+				'alt'   => 'Green Leaves',
+			)
+		);
+
+		// Run filter prior to Picture filter "webp_uploads_wrap_image_in_picture".
+		add_filter(
+			'wp_content_img_tag',
+			static function (
+				string $content
+			): string {
+				$processor = new WP_HTML_Tag_Processor( $content );
+				if ( ! $processor->next_tag( array( 'tag_name' => 'img' ) ) ) {
+					return $content;
+				}
+				$processor->set_attribute( 'sizes', '(max-width: 333px) 100vw, 333px' );
+				return $processor->get_updated_html();
+			}
+		);
```

Comment:
This is the only change to this test compared with the previous test, correct? As such, maybe better to use a data provider and have this `add_filter()` call be inside of a closure that is provided as an optional callback that gets executed at this point:

```php
function data_provider_test_image_sizes_equality(): array {
	return array(
		'no_filter' => array(
			'add_filter' => null,
		),
		'with_filter' => array(
			'add_filter' => static function () {
				add_filter(
					'wp_content_img_tag',
					static function (
						string $content
					): string {
						$processor = new WP_HTML_Tag_Processor( $content );
						if ( ! $processor->next_tag( array( 'tag_name' => 'img' ) ) ) {
							return $content;
						}
						$processor->set_attribute( 'sizes', '(max-width: 333px) 100vw, 333px' );
						return $processor->get_updated_html();
					}
				);
			},
		)
	);
}
```

So in the test you can just do:

```php
if ( $add_filter instanceof Closure ) {
	$add_filter();
}
```